### ISSUE TITLE
On Windows, can path complete handle forward and back slash?

### ISSUE THREAD

**Issue by electric-ladyland** [lifepillar/vim-mucomplete#78]
On Windows, path completion works with \\ (back slash). But in many files (especially code) it's customary to use / (forward slash). Is path/file completion able to handle both situations? Thanks.

**Comment by lifepillar** [comment]
The `'file'` completion method is supposed to work exactly as Vim native file completion (`<c-x><c-f>`), so it has the same limitations. For the `'path'` method, it might be possible. Since I do not use Windows, though, I need more information. How would you want it to work, exactly? Can you provide a few examples?

**Comment by electric-ladyland** [comment]
Sure. TypeScript is a good example. We often import files relative to the current file like this:
`import { ErrorHandler } from './error-handler/error-handler';`
If I were to start typing this out as is, when I type './e' I get no suggestions. However if I type '.\\e' it suggests the directory etc. Basically '\\' completion works exactly as I want, but rarely would I ever use this in my files (especially if this is a shared project) (not even sure if '\\' is legal). 

It's a similar situation in Java, SCSS, JS, to name a few I know if. Seems the trend is for source code to use '/' as separator, and handle the '\\' vs. '/' at a lower level so that source code remains independent of OS (?) So only matching '\\' on Windows is catering to a much smaller use case I would expect.

So basically, the current completion functionality is almost exactly what I want, except I want it to not only trigger on '\\' but on '/' also and complete using that as I type my path. I would never mix '\\' and '/' in one path. I develop exclusively on Windows and rarely ever type paths with '\\' (except for maybe some scripting) so personally it would be sufficient to be able to just hardcode which filesep to use in completion (but not change the Vim system filesep!) but some more intelligent handling of '\\' or '/' would be preferred.

**Comment by lifepillar** [comment]
Ok, it's clear. I'll try to implement it. Stay tuned.

**Comment by lifepillar** [comment]
Just wondering: have you tried setting `shellslash` (`:help shellslash`)? You should be able to type backslashes as usual, but have them changed to forward slashes by Vim.

**Comment by janlazo** [comment]
`set shellslash` is meant for `bash` and similar shells. It breaks `shellescape` for cmd.exe.

**Comment by lifepillar** [comment]
Ah ok, I misinterpreted the documentation.

**Comment by lifepillar** [comment]
@electric-ladyland Can you try the current master? Now you should be able to complete paths with slashes. There are still a few things to fix, but it should mostly work.
